src/components/find-on-shelf/FindOnShelfModal.tsx

```diff
@@ -1,41 +1,57 @@
 import * as React from "react";
 import { FC } from "react";
-import { ManifestationsSimpleFieldsFragment } from "../../core/dbc-gateway/generated/graphql";
+import { totalAvailableMaterialsInBranch } from "../../apps/material/helper";
+import {
+  ManifestationsSimpleFieldsFragment,
+  WorkMediumFragment
+} from "../../core/dbc-gateway/generated/graphql";
 import { useGetHoldingsV3 } from "../../core/fbs/fbs";
 import {
   convertPostIdToFaustId,
   creatorsToString,
   filterCreators,
-  flattenCreators
+  flattenCreators,
+  getManifestationsPids
 } from "../../core/utils/helpers/general";
 import Modal from "../../core/utils/modal";
 import { useText } from "../../core/utils/text";
 import { FaustId, Pid } from "../../core/utils/types/ids";
 import Disclosure from "../material/disclosures/disclosure";
+import FindOnShelfManifestationList from "./FindOnShelfManifestationList";
 
 export const findOnShelfModalId = (faustId: FaustId) =>
   `find-on-shelf-modal-${faustId}`;
 
 export interface FindOnShelfModalProps {
-  manifestation: ManifestationsSimpleFieldsFragment;
+  manifestations: ManifestationsSimpleFieldsFragment[];
+  workTitles: string[];
+  authors: WorkMediumFragment["creators"];
+  pid?: Pid;
```

Comment:
1. Great! Didn't know that Mikkel made these types 💯 🥇 
2. We recently got an email from the API devs saying that a work can indeed have multiple titles. In that case we just show them all separated by comma. 
3. I made it into Work["creators"] to support the new types, but I think we would prefer to have it this way over string[]. First of all because it's an array of objects, but also we use the authors prop later on in the code in filterCreators(authors, ["Person"]) function and the function doesn't accept array of strings, but rather array of creators objects - to match the data structure. And I think that's perfectly accurate. 
4. It is indeed for creating the modal ID. How would you approach handling this differently? :)